crates/http/src/server.rs

```diff
@@ -40,6 +41,14 @@ pub struct HttpServer<F> {
     #[cfg(feature = "http3")]
     #[cfg_attr(docsrs, doc(cfg(feature = "http3")))]
     enable_http3: bool,
+    /// Callback to configure socket used for http1 and http2
+    #[cfg(any(feature = "http1", feature = "http2"))]
+    #[cfg_attr(docsrs, doc(cfg(any(feature = "http1", feature = "http2"))))]
+    configure_h12_sock: Option<ConfigureSocketCallback>,
```

Comment:
can we make bon use an `Into` impl for this and then impl this for any `Fn<...>`